x/posts/internal/types/post_response.go

```diff
@@ -7,13 +7,34 @@ import (
 // PostQueryResponse represents the data of a post
 // that is returned to user upon a query
 type PostQueryResponse struct {
-	Post
+	Type      string    `json:"type"`
+	Post      Post      `json:"post"`
 	Reactions Reactions `json:"reactions"`
 	Children  PostIDs   `json:"children"`
 }
 
+func (response PostQueryResponse) String() string {
+	bytes, err := json.Marshal(&response)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bytes)
+}
+
 func NewPostResponse(post Post, reactions Reactions, children PostIDs) PostQueryResponse {
+	var ttype string
```

Comment:
I would prefer we rename this something like `msgType` 

```suggestion
	var msgType string
```

Also, we should initialize it to `TextPost` by default, and override the initial value later